src/persist/src/s3.rs

```diff
@@ -30,6 +31,24 @@ impl Config {
     #[cfg(test)]
     const EXTERNAL_TESTS_S3_BUCKET: &'static str = "MZ_PERSIST_EXTERNAL_STORAGE_TEST_S3_BUCKET";
 
+    /// Returns a new [Config] for use in production.
+    ///
+    /// Stores objects in the given bucket prepended with the (possibly empty)
+    /// prefix. S3 credentials and region must be available in the process or
+    /// environment.
+    pub fn new(bucket: String, prefix: String) -> Result<Self, Error> {
```

Comment:
@benesch looking for extra careful review on this plz

Reply:
Yeah, idk, it's probably a good enough place to start. The Rusoto credentials chain is missing some credential providers, but that's not really fixable here. If users want to explicitly configure the access keys they can do so via the `AWS_ACCESS_KEY_ID` environment variables.